lua/apisix/discovery/eureka.lua

```diff
@@ -0,0 +1,181 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local local_conf         = require("apisix.core.config_local").local_conf()
+local service            = require("apisix.discovery.service")
+local http               = require("resty.http")
+local core               = require("apisix.core")
+local ipairs             = ipairs
+local ngx_timer_at       = ngx.timer.at
+local ngx_timer_every    = ngx.timer.every
+local string_sub         = string.sub
+local string_find        = string.find
+local log                = core.log
+
+local applications
+local useragent = 'ngx_lua-apisix/v' .. core.version.VERSION
+
+local _M = {
+    version = 1.0,
+}
+
+
+local function split(self, sep)
+    local sep, fields = sep or ":", {}
+    local pattern = string.format("([^%s]+)", sep)
+    self:gsub(pattern, function(c)
+        fields[#fields + 1] = c
+    end)
+    return fields
+end
+
+
+local function service_info()
+    if not local_conf.eureka or not local_conf.eureka.urls then
+        log.error("do not set eureka.url")
+        return
+    end
+    local service_url = local_conf.eureka.urls
+    local urls = split(service_url, [[,]])
+    local basic_auth
+    local url = urls[math.random(#urls)]
+    local user_and_password_idx = string_find(url, "@")
```

Comment:
https://www.lua.org/manual/5.1/manual.html#pdf-string.find

we'd better use the plain model which is faster.